torch/testing/_internal/common_quantization.py

```diff
@@ -474,7 +474,8 @@ def _group_quantize_tensor(w, n_bit=4, q_group_size=16):
     assert torch.isnan(out).sum() == 0
 
     out = out.to(dtype=torch.int32).reshape(w.shape)
-    out_uint8 = (out[::, ::2] << 4 | out[::, 1::2]).to(torch.uint8)
+    if out.device != torch.device('cpu'):
+        out = (out[::, ::2] << 4 | out[::, 1::2]).to(torch.uint8)
```

Comment:
OK so cuda op now has changed format `[n, k/2]` is that correct?

I guess it might be too disruptive to change it back, so just asking here

Reply:
Now, only CPU format is changed, and CUDA/MPS format remains the same as what I did in https://github.com/pytorch/pytorch/pull/129940.

Do you have the request of changing CUDA/MPS format from [n, k / 2] uint8 back to [n, k] int32? cc @mingfeima

Reply:
not right now, I just want to confirm my understanding. changing back would be another round of breaking changes so I think we can wait until it's strictly required